### PR TITLE
Feat/question service

### PR DESCRIPTION
**Pull request by Josephaedan**
Adds question-service server to monorepo
- There are 2 environments for pulumi under [cs3219/question-service-infra](https://app.pulumi.com/cs3219/question-service-infra)
- Staging URL: https://questions.staging.peerprep.net
- Prod URL: https://questions.peerprep.net
- In Prod/Staging, the MongoDB URL references an externally hosted instance, which will be updated to an Mongo Atlas instance spun up in the repo later
- In Local Development (```docker-compose up```), the MongoDB instance will be spun up in a local container containing base questions for testing.

### CODE REVIEW

File: backend/question-service/server/app/server.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    
+    if difficulty:
+        query_filter["difficulty"] = difficulty
+    if topic:
+        query_filter["topic_tags"] = topic
+    if language:
+        query_filter["languages"] = language
+
+    cursor = collection.find(query_filter, {"_id": 0, "id": 1, "title": 1})
```
Comment (raynerljm) [inline backend/question-service/server/app/server.py]: what does this line of code mean? specifically the `"id": 1, "title": 1` thing

File: backend/question-service/server/app/server.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    questions = list(cursor)
+
+    return jsend_response("success", {"questions": questions})
```
Comment (raynerljm) [inline backend/question-service/server/app/server.py]: nit: is it possible to abstract "success" and "fail" into an enum - idk if there is smth like that in python

File: backend/question-service/server/infra/Pulumi.prod.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+config:
+  aws:region: ap-southeast-1
+  question-service-infra:MONGO_HOST:
+    secure: AAABACsWLU0lTfORYPsnr2nzFVQ3eHH+crdTZBGoHnhg2ryS5uOAELHuFHovd8Pd9CdeEvHIBlSiiJwZPpDeeZCnsn7t4TUEI6QkUP5cDUfr4UlbZ1h/O2/lLUU5
+  question-service-infra:MONGO_PORT:
```
Comment (raynerljm) [inline backend/question-service/server/infra/Pulumi.prod.yaml:5]: nit: actl idt the port needs to be a secret, I think can be stored in plaintext

File: docker-compose.yml [hunk 4]
```diff
@@ -20,6 +20,34 @@ services:
...
+        condition: service_healthy
+    ports:
+      - "8003:80"
+  
+  question-service-db:
+    image: mongo:latest
+    volumes:
+      - question-service-db-data:/data/db
+    ports:
+      - "27017:27017"
```
Comment (raynerljm) [inline docker-compose.yml:44]: i think this is fine for now but we need to eventually find a way to deconflict with other databases/ find another arrangement

**Review by raynerljm — APPROVED** [review]
lgtm! just some nits

File: backend/question-service/server/app/server.py (hunk 2 above)
Comment (Josephaedan) [reply backend/question-service/server/app/server.py]: Added Enum type for JSendStatus

File: backend/question-service/server/app/server.py (hunk 1 above)
Comment (Josephaedan) [reply backend/question-service/server/app/server.py]: Added Project Fields to return id and title

**Pull request merged**